src/sentry/middleware/integrations/integration_control.py

```diff
@@ -1,100 +1,48 @@
 from __future__ import annotations
 
 import logging
-import re
-from typing import Callable, Mapping, Type
+from typing import Callable, List, Type
 
 from django.http import HttpRequest, HttpResponse
 
-from sentry.silo import SiloMode
-
-from .parsers import (
-    BitbucketRequestParser,
-    BitbucketServerRequestParser,
-    GithubEnterpriseRequestParser,
-    GithubRequestParser,
-    GitlabRequestParser,
-    JiraRequestParser,
-    JiraServerRequestParser,
-    MsTeamsRequestParser,
-    SlackRequestParser,
-    VstsRequestParser,
+from sentry.middleware.integrations.classifications import (
+    BaseClassification,
+    IntegrationClassification,
 )
-from .parsers.base import BaseRequestParser
+from sentry.silo import SiloMode
 
 logger = logging.getLogger(__name__)
 
-ACTIVE_PARSERS = [
-    BitbucketRequestParser,
-    BitbucketServerRequestParser,
-    GithubEnterpriseRequestParser,
-    GithubRequestParser,
-    GitlabRequestParser,
-    JiraRequestParser,
-    JiraServerRequestParser,
-    MsTeamsRequestParser,
-    SlackRequestParser,
-    VstsRequestParser,
-]
-
 
 class IntegrationControlMiddleware:
     integration_prefix: str = "/extensions/"
     """Prefix for all integration requests. See `src/sentry/web/urls.py`"""
     setup_suffix: str = "/setup/"
     """Suffix for PipelineAdvancerView on installation. See `src/sentry/web/urls.py`"""
 
-    integration_parsers: Mapping[str, Type[BaseRequestParser]] = {
-        parser.provider: parser for parser in ACTIVE_PARSERS
-    }
+    classifications: List[Type[BaseClassification]] = [
+        IntegrationClassification,
+    ]
+    """Classifications to determine whether request must be parsed, sorted in priority order."""
 
     def __init__(self, get_response: Callable[[], HttpResponse]):
         self.get_response = get_response
 
-    def _identify_provider(self, request: HttpRequest) -> str | None:
-        """
-        Parses the provider out of the request path
-            e.g. `/extensions/slack/commands/` -> `slack`
-        """
-        integration_prefix_regex = re.escape(self.integration_prefix)
-        provider_regex = rf"^{integration_prefix_regex}(\w+)"
-        result = re.search(provider_regex, request.path)
-        if not result:
-            logger.error(
-                "integration_control.invalid_provider",
-                extra={"path": request.path},
-            )
-            return None
-        return result.group(1)
-
     def _should_operate(self, request: HttpRequest) -> bool:
         """
         Determines whether this middleware will operate or just pass the request along.
         """
         is_correct_silo = SiloMode.get_current_mode() == SiloMode.CONTROL
-        is_integration = request.path.startswith(self.integration_prefix)
-        is_not_setup = not request.path.endswith(self.setup_suffix)
-        return is_correct_silo and is_integration and is_not_setup
+        return is_correct_silo
```

Comment:
As is, this will operate on every request received at the control silo, and then each classification will check if they should operate or not, returning early if it does. 

We may be able to reduce the number of these checks if we narrow down similarities between integration requests, plugin requests and provisioning requests, but currently, that is not being done.

Reply:
We should be ok. The `should_operate` methods are both simple and should be low overhead.